### PR TITLE
Austenem/CAT-929 double dialog bug

### PR DESCRIPTION
**Pull request by austenem**
## Summary

Fixes a bug present on the dataset detail pages. Previously, the "Add to Workspace" dialog could not be saved, and opened two dialogs simultaneously.

## Design Documentation/Original Tickets

[CAT-929 Jira ticket](https://hms-dbmi.atlassian.net/browse/CAT-929?atlOrigin=eyJpIjoiMmNkMzY5N2VmMzUzNGQ4NGFkNjRkYzkyNWZmN2E0MDQiLCJwIjoiaiJ9)

## Testing

Tested manually that dialogs worked as expected for both the entity header and the helper menu workspace options. Also tested that search page dialog continued to work as expected.

## Screenshots/Video

Issue present on prod:

https://github.com/user-attachments/assets/f5f2309f-ee5f-4a2d-b7dc-ab2812d85faf

Resolution:

https://github.com/user-attachments/assets/7d02e5bc-941c-46e8-8116-dfaee1ed493c

## Checklist

- [X] Code follows the project's coding standards
    - [X] Lint checks pass locally
    - [X] New `CHANGELOG-your-feature-name-here.md` is present in the root directory, describing the change(s) in full sentences.
- [X] Unit tests covering the new feature have been added
- [X] All existing tests pass
- [X] Any relevant documentation in JIRA/Confluence has been updated to reflect the new feature
- [X] Any new functionalities have appropriate analytics functionalities added

## Additional Notes

For clarity, the cause of the bug was that there were two "Add dataset to workspace" dialogs - one for the helper menu, and one for the entity header - and both depended on a single boolean to determine whether they were open. As a result, whenever one was opened, the other would open behind it. 

This was resolved by using one dialog for both the helper menu and the entity header and storing the relevant dataset to use in the dialog, which also resolved the issue of the dialog not responding.

### CODE REVIEW

File: context/app/static/js/pages/Dataset/Dataset.tsx [hunk 1]
```diff
@@ -26,9 +22,17 @@ import DatasetRelationships from 'js/components/detailPage/DatasetRelationships'
...
 import SupportAlert from 'js/components/detailPage/SupportAlert';
-import OrganIcon from 'js/shared-styles/icons/OrganIcon';
-import Stack from '@mui/material/Stack';
+import { useTrackEntityPageEvent } from 'js/components/detailPage/useTrackEntityPageEvent';
 import { useDatasetRelationships } from 'js/components/detailPage/DatasetRelationships/hooks';
+import AddDatasetsFromSearchOrDetailDialog from 'js/components/workspaces/AddDatasetsFromSearchOrDetailDialog';
+
+import { useDatasetsCollections } from 'js/hooks/useDatasetsCollections';
+import useTrackID from 'js/hooks/useTrackID';
+
```
Comment (john-conroy) [inline context/app/static/js/pages/Dataset/Dataset.tsx:31]:
```suggestion
```

File: context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx [hunk 2]
```diff
@@ -56,7 +55,7 @@ function ProcessedDataWorkspaceMenu({
     initialSelectedDatasets: [uuid].filter(Boolean),
   });
 
-  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH');
+  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH', uuid);
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx:58]: I'm a little confused as to why we need to pass uuid here, also to the `SelectableTableProvider` in this component and include a `SelectableTableProvider` without the `uuid` in the dataset component.

File: context/app/static/js/pages/Dataset/Dataset.tsx [hunk 3]
```diff
@@ -125,36 +129,39 @@ function DatasetDetail({ assayMetadata }: EntityDetailProps<Dataset>) {
...
-          <MetadataSection shouldDisplay={shouldDisplaySection.metadata} />
-          <ProcessedDataSection shouldDisplay={Boolean(shouldDisplaySection['processed-data'])} />
-          <BulkDataTransfer shouldDisplay={Boolean(shouldDisplaySection['bulk-data-transfer'])} />
-          <ProvSection shouldDisplay={shouldDisplaySection.provenance} />
-          <CollectionsSection shouldDisplay={shouldDisplaySection.collections} />
-          <Attribution>
-            <ContributorsTable contributors={contributors} contacts={contacts} />
-          </Attribution>
-        </DetailLayout>
+        <SelectableTableProvider tableLabel="Current Dataset">
```
Comment (john-conroy) [inline context/app/static/js/pages/Dataset/Dataset.tsx:132]: Does the table provider need to wrap everything or just the `AddDatasetsFromSearchOrDetailDialog`?

File: context/app/static/js/components/workspaces/AddDatasetsFromSearchOrDetailDialog/hooks.ts [hunk 4]
```diff
@@ -139,20 +143,19 @@ function useAddDatasetsFromSearchDialog() {
     [updateWorkspaceDatasets, workspaceDatasets],
   );
 
-  const { isOpen } = useEditWorkspaceStore();
   // react-hook-form's defaultValues are cached and must be set upon open. https://react-hook-form.com/docs/useform#defaultValues
   useEffect(() => {
-    setValue('datasets', datasetsFromSearch);
-  }, [datasetsFromSearch, setValue, protectedHubmapIds, isOpen]);
+    setValue('datasets', datasetsToUse);
+  }, [datasetsToUse, setValue, protectedHubmapIds]);
```
[the remark this answers is not in the document]
Comment (austenem) [reply context/app/static/js/components/workspaces/AddDatasetsFromSearchOrDetailDialog/hooks.ts]: This was accidental, good catch!

File: context/app/static/js/components/detailPage/ProcessedData/HelperPanel/HelperPanel.tsx [hunk 5]
```diff
@@ -110,7 +111,9 @@ function HelperPanelActions() {
           </SecondaryBackgroundTooltip>
         }
         datasetDetails={{ hubmap_id, uuid, status }}
+        dialogType="ADD_DATASETS_FROM_HELPER_PANEL"
       />
+      <AddDatasetsFromDetailDialog uuid={uuid} dialogType="ADD_DATASETS_FROM_HELPER_PANEL" />
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/ProcessedData/HelperPanel/HelperPanel.tsx:116]: Can we just move this into `ProcessedDataWorkspaceMenu`?

**Review by john-conroy — COMMENTED** [review]
Looks great! Just one small comment.

File: context/app/static/js/components/detailPage/ProcessedData/HelperPanel/HelperPanel.tsx (hunk 5 above)
Comment (austenem) [reply context/app/static/js/components/detailPage/ProcessedData/HelperPanel/HelperPanel.tsx]: Yes!

**Review by john-conroy — APPROVED** [review]
Great work! Merge once CI passes.

**Pull request merged**